ui/components/app/signature-request-siwe/signature-request-siwe.container.js

```diff
@@ -0,0 +1,41 @@
+import { connect } from 'react-redux';
+import { accountsWithSendEtherInfoSelector } from '../../../selectors';
+import { getAccountByAddress } from '../../../helpers/utils/util';
+import SignatureRequest from './signature-request-siwe';
+
+function mapStateToProps(state) {
+  const {
+    metamask: { subjectMetadata = {} },
```

Comment:
There's no need for this container file since we are using functional components. We can access redux using useSelector. We can also shave off some complexities by working with the passed in params, `ownProps`, directly in the component file itself. 

After deleting this container file, we can do something like 

```
import { useSelector } from 'react-redux';
import {
  accountsWithSendEtherInfoSelector,
  getSubjectMetadata,
} from '../../../selectors';
import { getAccountByAddress } from '../../../helpers/utils/util';

export default function SignatureRequestSIWE({
  txData,
  cancelPersonalMessage,
  signPersonalMessage,
}) {
  // ...

  const allAccounts = useSelector(accountsWithSendEtherInfoSelector);
  const subjectMetadata = useSelector(getSubjectMetadata);

  const {
    msgParams: {
      from,
      origin,
      siwe: { isSIWEDomainValid, isMatchingAddress, messageData },
      version,
    },
    type,
  } = txData;

  const fromAccount = getAccountByAddress(allAccounts, from);
  const targetSubjectMetadata = subjectMetadata[origin];
```

Reply:
Addressed!